src/cljdoc/doc_tree.clj

```diff
@@ -30,6 +30,7 @@
 (spec/def :cljdoc.doc/type #{:cljdoc/markdown :cljdoc/asciidoc})
 (spec/def :cljdoc/asciidoc string?)
 (spec/def :cljdoc.doc/contributors (spec/coll-of string?))
+(spec/def :cljdoc.doc/external-url (spec/and ::ne-string #(try (java.net.URI. %) true (catch Exception _ false))))
```

Comment:
You're right that this accepts a lot of different URLs that aren't particularly helpful. I think we might want to do further checking beyond this. Maybe we should just [check for a scheme of http/https](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#getScheme--) (case insensitive) and [a hostname that's not empty](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#getHost--)? Everything else should be optional, really? That seems like a really simple change, it could be thrown in a function.

Also, this should only catch `URISyntaxException`, just so that we're not accidentally swallowing other exceptions.

Reply:
FWIW I think in the interest of unblocking this contribution I'd be totally fine with this approach. It might not be perfect but it's certainly an improvement. In a tradeoff between making the contribution process easy and fun and perfection I usually tend to chose the former — which may or may not be a good decision down the line 😂

Reply:
You definitely don't have to worry about the all the comments. I want to become a better Clojure dev and I appreciate all the feedback I get here.